beeline-core/src/main/java/io/honeycomb/beeline/tracing/README.md

```diff
@@ -0,0 +1,144 @@
+# Instrumenting calls
+
+## List of contents
+* [Apache HTTP Components (Synchronous)](#apache-synchronous)
+* [Apache HTTP Components (Async)](#apache-async)
+* [Jersey](#jersey)
+* [Java HTTP client (Synchronous)](#synchronous)
+* [Java HTTP client (Asynchronous)](#asynchronous)
+* [Resteasy](#resteasy)
+* [Dropwizard](#dropwizard)
+
+## Client Calls
+Adding honeycomb trace header can help you track calls from one system to the next. Each HTTP client framework will have a slightly different way to add the header. To have this done automatically for each request, you will typically create a custom Interceptor
+### Apache HTTP Components
+#### Apache Synchronous
+[Apache's HttpComponents Client](https://hc.apache.org/httpcomponents-client-4.5.x/index.html)
+
+Example creating request
+```java
+// create HTTP client
+CloseableHttpClient client = HttpClients.custom().setConnectionTimeToLive(30, TimeUnit.SECONDS).build();
+// Create HTTP Request using GET
+HttpGet getRequest = new HttpGet("https://google.com");
+// Add Honeycomb trace header
+getRequest.addHeader("x-honeycomb-trace", MessageFormat.format("1;trace_id={0},parent_id={1}", span.getTraceId(), span.getParentSpanId()));
+// Execute HTTP request and get response
+CloseableHttpResponse getResponse = client.execute(getRequest);
+// Create HTTP Request using POST
+HttpPost postRequest = new HttpPost("");
+// Add Honeycomb trace header
+postRequest.addHeader("x-honeycomb-trace", MessageFormat.format("1;trace_id={0},parent_id={1}", span.getTraceId(), span.getParentSpanId()));
+// Execute HTTP request and get response
+CloseableHttpResponse postResponse = client.execute(postRequest);
+```
+
+Use the HttpClientPropagator and a custom RequestExecutioner to programmatically handle this.
+#### Apache Async
+[Apache's HttpComponents Async Client](https://hc.apache.org/httpcomponents-asyncclient-4.1.x/index.html)
+
+Example creating request with Honeycomb Trace set. This example uses the returned future for processing.
+```java
+final HttpGet request = new HttpGet("http://google.com");
+request.addHeader("x-honeycomb-trace", MessageFormat.format("1;trace_id={0},parent_id={1}", span.getTraceId(), span.getParentSpanId()));
+final Future<HttpResponse> responseFuture = client.execute(request, null, null);
+final HttpResponse response = responseFuture.get();
+// process response here
+```
+
+Example creating request with Honeycomb Trace set. This example uses a callback for processing.
+```java
+final HttpGet request = new HttpGet("http://google.com");
+request.addHeader("x-honeycomb-trace", MessageFormat.format("1;trace_id={0},parent_id={1}", span.getTraceId(), span.getParentSpanId()));
```

Comment:
Isn't this a constant in `HttpHeaderV1PropagationCodec`? These examples require the user to know the format of the header - is there a way to encode the context using that codec class?